plonk/src/circuit/basic.rs

```diff
@@ -261,6 +261,12 @@ impl<F: FftField> PlonkCircuit<F> {
     pub fn range_size(&self) -> Result<usize, PlonkError> {
         Ok(1 << self.range_bit_len()?)
     }
+
+    /// The required SRS size for the circuit.
+    pub fn srs_size(&self) -> Result<usize, PlonkError> {
+        // extra 2 degree for masking polynomial to make snark zero-knowledge
+        Ok(self.num_gates() + 2)
+    }
```

Comment:
- I personally don't know which one of `srs_size()` or `max_degree()` is a better name
- should we add this to `trait Arithmetization`?
- We should also update code here: https://github.com/SpectrumXYZ/jellyfish/blob/main/plonk/src/proof_system/snark.rs#L76-L81

(cc @chancharles92 for input)

Reply:
> We should also update code here: https://github.com/SpectrumXYZ/jellyfish/blob/main/plonk/src/proof_system/snark.rs#L76-L81

Yeah... seems that that block of code needs its own API too -- input an SRS and a circuit, return whether if the srs is sufficient for that circuit.

Reply:
Agree with @alxiong on point 2 and 3, for point 1, I'm fine with either name.

@zhenfeizhang I don't think it's necessary to have a new API, maybe just write:
``if srs.0.max_degree() < circuit.srs_size() {
            return Err(PlonkError::IndexTooLarge);
   }``